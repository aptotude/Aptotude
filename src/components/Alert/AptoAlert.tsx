import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoAlert.scss';

const COMPONENT_PREFIX = 'AptoAlert';

export interface AptoAlertDisplayProps extends StandardTypes {
  variant?: 'info' | 'success' | 'warning' | 'danger';
}

export class AptoAlert extends React.Component<AptoAlertDisplayProps> {
  public static defaultProps = {
    variant: 'info'
  };

  public render() {
    const { variant, className, forwardRef, children, ...rest } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      variant && `${COMPONENT_PREFIX}--${variant}`,
      className
    );

    return (
      <div {...rest} ref={forwardRef} className={classes}>
        {children}
      </div>
    );
  }
}
