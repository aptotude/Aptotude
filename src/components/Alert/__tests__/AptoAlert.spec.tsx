import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoAlert } from '../AptoAlert';

afterEach(() => {
  cleanup();
});

describe('AptoAlert', () => {
  it('Renders Component', () => {
    const { container } = render(<AptoAlert>scary</AptoAlert>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders info', () => {
    const { container } = render(<AptoAlert variant="info">scary</AptoAlert>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders success', () => {
    const { container } = render(
      <AptoAlert variant="success">scary</AptoAlert>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders warning', () => {
    const { container } = render(
      <AptoAlert variant="warning">scary</AptoAlert>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders danger', () => {
    const { container } = render(<AptoAlert variant="danger">scary</AptoAlert>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders extra classes', () => {
    const { container } = render(<AptoAlert className="foo">scary</AptoAlert>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
