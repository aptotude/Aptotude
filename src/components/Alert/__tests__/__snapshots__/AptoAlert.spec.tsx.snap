// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoAlert Renders Component 1`] = `
<div
  class="AptoAlert AptoAlert--info"
>
  scary
</div>
`;

exports[`AptoAlert Renders danger 1`] = `
<div
  class="AptoAlert AptoAlert--danger"
>
  scary
</div>
`;

exports[`AptoAlert Renders extra classes 1`] = `
<div
  class="AptoAlert AptoAlert--info foo"
>
  scary
</div>
`;

exports[`AptoAlert Renders info 1`] = `
<div
  class="AptoAlert AptoAlert--info"
>
  scary
</div>
`;

exports[`AptoAlert Renders success 1`] = `
<div
  class="AptoAlert AptoAlert--success"
>
  scary
</div>
`;

exports[`AptoAlert Renders warning 1`] = `
<div
  class="AptoAlert AptoAlert--warning"
>
  scary
</div>
`;
